astropy/io/misc/asdf/tags/transform/polynomial.py

```diff
@@ -274,15 +276,25 @@ def to_tree_transform(cls, model, ctx):
                     coefficients[i, j] = getattr(model, name).value
         node = {'polynomial_type': poly_type, 'coefficients': coefficients}
         if ndim == 1:
-            if model.domain is not None:
+            if model.domain != model._default_domain_window['domain']:
                 node['domain'] = model.domain
-            if model.window is not None:
+            if model.window != model._default_domain_window['window']:
                 node['window'] = model.window
         else:
-            if model.x_domain or model.y_domain is not None:
-                node['domain'] = (model.x_domain, model.y_domain)
-            if model.x_window or model.y_window is not None:
-                node['window'] = (model.x_window, model.y_window)
+            domain = []
+            window = []
+            if model.x_domain != model._default_domain_window['x_domain']:
+                domain.append(model.x_domain)
+            if model.y_domain != model._default_domain_window['y_domain']:
+                domain.append(model.y_domain)
+            if model.x_window != model._default_domain_window['x_window']:
+                window.append(model.x_window)
+            if model.y_window != model._default_domain_window['y_window']:
+                window.append(model.y_window)
```

Comment:
Does this do the right thing if only one of the two x or y values passes their test? It looks like if either of the x or y values equals its default, it doesn't get appended, leaving a list with only one value.

Reply:
Only the one that is non-default is written to file.
DO you think we should write out the default value as well?

Reply:
I may be missing the rest of the context, but it looks like there is no information in domain or window to indicate which coordinate the domain is associated with. If there is only one domain for a 2d case, is it for x or y?

Reply:
Nope, I missed that. I'll write them out separately.

Reply:
This is fixed now but let's wait for tests. I noticed, unrelated, that default values of constraints were serialized. Fixed it in this PR.